lib/modules/datasource/custom/formats/toml.ts

```diff
@@ -0,0 +1,23 @@
+import { readLocalFile } from '../../../../util/fs';
+import type { Http } from '../../../../util/http';
+import { parse as parseToml } from '../../../../util/toml';
+import type { CustomDatasourceFetcher } from './types';
+
+export class TomlFetcher implements CustomDatasourceFetcher {
+  async fetch(http: Http, registryURL: string): Promise<unknown> {
+    const response = await http.getToml(registryURL);
+
+    const contentType = response.headers['content-type'];
+    if (!contentType?.startsWith('application/toml')) {
+      return null;
+    }
+
+    return parseToml(response.body);
+  }
+
+  async readFile(registryURL: string): Promise<unknown> {
+    const fileContent = await readLocalFile(registryURL, 'utf8');
+
+    return parseToml(fileContent!);
+  }
```

Comment:
we should define this in a follow up PR. so we've a read and a parse function in base class for less duplicate code